test/test_tensorexpr.py

```diff
@@ -1065,6 +1066,43 @@ def easy(x, y):
         # FIXME: interp.elapsed_value() also increments due to simplifier
         assert llvm.elapsed_value() == 1 or interp.elapsed_value() > 1
 
+    def _test_softmax(self, device):
+        @torch.jit.script
+        def test(x, y):
+            a = F.softmax(x, dim=0)
+            b = F.softmax(y, dim=0)
+            c = F.softmax(x, dim=1)
+            d = F.softmax(y, dim=1)
+            return a + b + c + d
+
+        def test_ref(x, y):
```

Comment:
This PR reminds me that we should stop using numpy/scipy as a reference and use torch eager mode instead...

Reply:
What is the recommendation / expectation of the community? Isn't it expected that our ops try to match the semantics of those in numpy / scipy? If thats the case, wouldn't it better to stick to them as the reference points?

Reply:
If we diverge from eager mode, then it's a bug in our code and we should fix it. If the eager mode diverges from numpy/scipy, we should raise this issue, but it's not always a must-fix bug, and it is not blocking *TE* changes.

Reply:
Changed it to use eager mode as ref.